File: samples/java/mtSampleApp/src/main/java/com/rws/lt/lc/mtsampleapp/exception/AppException.java
package com.rws.lt.lc.mtsampleapp.exception;

import com.rws.lt.lc.mtsampleapp.transfer.ErrorDetail;
import lombok.Getter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Getter
@ToString(callSuper = true)
public class AppException extends RuntimeException {

    private final String errorCode;
    protected List<ErrorDetail> details = new ArrayList<>();

    public AppException(String errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }

    public AppException(String errorCode, String message, List<ErrorDetail> errorDetails) {
        super(message);
        this.errorCode = errorCode;
        this.details = errorDetails;
    }

    public void addError(ErrorDetail propertyValidationError) {
        this.details.add(propertyValidationError);
    }
}

File: samples/dotnet/MTSampleApp/Rws.LC.MTSampleApp/Exceptions/AppException.cs
using System;
using System.Net;

namespace Rws.LC.MTSampleApp.Exceptions
{
    public class AppException : Exception
    {
        public string ErrorCode { get; set; }
        public HttpStatusCode StatusCode { get; set; }
        public Details[] ExceptionDetails { get; set; }

        public AppException()
        {
            ErrorCode = ErrorCodes.InternalServerError;
            StatusCode = HttpStatusCode.InternalServerError;
        }

        public AppException(string message)
            : base(message)
        {
            ErrorCode = ErrorCodes.InternalServerError;
            StatusCode = HttpStatusCode.InternalServerError;
        }

        public AppException(string message, Exception inner)
            : base(message, inner)
        {
            ErrorCode = ErrorCodes.InternalServerError;
            StatusCode = HttpStatusCode.InternalServerError;
        }

        public AppException(HttpStatusCode statusCode, string errorCode, string message)
            : base(message)
        {
            ErrorCode = errorCode;
            StatusCode = statusCode;
        }

        public AppException(HttpStatusCode statusCode, string errorCode, string message, Details[] details)
            : base(message)
        {
            ErrorCode = errorCode;
            StatusCode = statusCode;
            ExceptionDetails = details;
        }

        public AppException(HttpStatusCode statusCode, string errorCode, string message, Exception inner)
            : base(message, inner)
        {
            ErrorCode = errorCode;
            StatusCode = statusCode;
        }

        public AppException(HttpStatusCode statusCode, string errorCode, string message, Details[] details, Exception inner)
            : base(message, inner)
        {
            ErrorCode = errorCode;
            StatusCode = statusCode;
            ExceptionDetails = details;
        }
    }
}
